crates/trigger-http/src/lib.rs

```diff
@@ -240,19 +249,25 @@ impl HttpTrigger {
             req.uri()
         );
 
-        let path = req.uri().path();
+        let path = req.uri().path().to_string();
 
         // Handle well-known spin paths
         if let Some(well_known) = path.strip_prefix(spin_http::WELL_KNOWN_PREFIX) {
             return match well_known {
-                "health" => Ok(Response::new(body::full(Bytes::from_static(b"OK")))),
-                "info" => self.app_info(),
+                "health" => {
+                    let mut response = Response::new(body::full(Bytes::from_static(b"OK")));
+                    response
+                        .extensions_mut()
+                        .insert(MatchedRoute { route: path });
+                    Ok(response)
+                }
+                "info" => self.app_info(path),
                 _ => Self::not_found(NotFoundRouteKind::WellKnown),
```

Comment:
Minor: you could probably avoid a little complexity by setting the extension out here, e.g.:

```rust
let mut resp = match well_known {
  ...
};
resp.extenstions_mut()...
return Ok(resp)
```
...though maybe that doesn't work well with the `NotFoundRouteKind::WellKnown`? :shrug:

Reply:
Correct, doesn't work b/c we don't want to set the route in the not found case.